CRM/Utils/System/Joomla.php

```diff
@@ -1069,4 +1080,37 @@ public function modifyStandaloneProfile($profile, $params):string {
     return $profile;
   }
 
+  protected function loadJoomlaApplication(string $applicationName) {
+    // Boot the DI container
+    if ($applicationName === 'site') {
+      $applicationClass = \Joomla\CMS\Application\SiteApplication::class;
+      $session = 'session.web.site';
+    }
+    elseif ($applicationName == 'admin') {
+      $applicationClass = \Joomla\CMS\Application\AdministratorApplication::class;
+      $session = 'session.web.administrator';
+    }
+    else {
+      $applicationClass = \Joomla\CMS\Application\ConsoleApplication::class;
+      $session = 'session.cli';
+    }
+    $container = \Joomla\CMS\Factory::getContainer();
+    /*
+     * Alias the session service keys to the web session service as that is the primary session backend for this application
+     *
+     * In addition to aliasing "common" service keys, we also create aliases for the PHP classes to ensure autowiring objects
+     * is supported.  This includes aliases for aliased class names, and the keys for aliased class names should be considered
+     * deprecated to be removed when the class name alias is removed as well.
+     */
+    $container->alias('session', $session)
+      ->alias('JSession', $session)
+      ->alias(\Joomla\CMS\Session\Session::class, $session)
+      ->alias(\Joomla\Session\Session::class, $session)
+      ->alias(\Joomla\Session\SessionInterface::class, $session);
+    // Instantiate the application.
+    $app = $container->get($applicationClass);
```

Comment:
I had to add `$app->createExtensionNamespaceMap();` after this line to get it to work. See [main comment](https://github.com/civicrm/civicrm-core/pull/32830#issuecomment-2931668133) for further discussion.